src/NBitcoin/ChainedHeader.cs

```diff
@@ -536,6 +536,28 @@ public ChainedHeader FindFork(ChainedHeader block)
             return highChain;
         }
 
+        /// <summary>
+        /// Convert the current and given chained headers  on the same chain to a list of consecutive items. 
+        /// </summary>
+        /// <param name="stopHeader">The header at the bottom.</param>
+        /// <returns>List of consecutive items.</returns>
+        /// <exception cref="InvalidOperationException">Then the items are not on the same chain.</exception>
+        public IEnumerable<ChainedHeader> ToConsecutiveList(ChainedHeader stopHeader)
```

Comment:
I don't think this method is very helpful. Seems to just adding overhead when you want to go down the chain. Its name also does not match what it does and the summary also does not match what it does.

Reply:
Either you do it here or in `ConnectNewHeadersResult` I dont mind, but it maybe useful to have it on the chain header.
If comment is incorrect then we can just fix the comment

Reply:
If you want to keep it, then yes, fix the name and fix the comments. But I'm questioning whether we need such a method at all.